deliver/lib/deliver/app_trailer.rb

```diff
@@ -0,0 +1,60 @@
+module Deliver
+  # AppTrailer represents one video trailer for one specific locale and
+  # device type.
+  class AppTrailer
+
+    attr_accessor :device_type
+
+    attr_accessor :video_path
+
+    attr_accessor :language
+
+    attr_accessor :timestamp
+
+    attr_accessor :preview_image_path
+
+    # @param video_path (String) path to the video file
+    # @param language (String) Language of this trailer (e.g. en-US)
+    def initialize(video_path, language)
+      self.video_path = video_path
+      self.language = language
+
+      # should be given as video file name prefix since we can't determine device by resolution
+      self.device_type = self.class.discover_device_type(video_path)
+      self.timestamp = '05.00'
+      # should be in the same directory as video file and also prefixed with device type
+      self.preview_image_path = self.class.discover_preview_image(video_path, device_type)
+
+      UI.user_error!("The extension of given video (#{video_path}) does not match the requirements (mov, m4v, mp4)") unless self.valid_file_extension?
+    end
+
+    # Validates file extension
+    def valid_file_extension?
+      return ["mov", "m4v", "mp4"].include?(self.video_path.split(".").last.downcase)
+    end
+
+    # Let's agree that device type is a prefix eg. iphone4_filename.mov
+    def self.discover_device_type(video_path)
+      filename = File.basename(video_path)
+      dev = filename.split('_').first
+      # Trailers available only for those device types
+      dev_types = ['iphone4', 'iphone6', 'iphone6Plus', 'ipad', 'ipadPro', 'appleTV']
+
+      UI.user_error!("Unrecognized device type for path '#{video_path}'") unless dev_types.include?(dev)
+      return dev
+    end
+
+    # Finds preview image matching to the given video, assuming that preview is also prefixed with device type
+    def self.discover_preview_image(video_path, device_type)
+      preview_path = File.dirname(video_path)
+      preview_extensions = '{png,jpg,jpeg}'
+
+      files = Dir.glob(File.join(preview_path, "*.#{preview_extensions}"), File::FNM_CASEFOLD)
+
+      files.each do |file|
+        return file if file.include?(device_type)
+      end
+      return nil
```

Comment:
I don't think we want to support not having the file poster image in the first implementation. So we want to fail here. `user_error!`